frontends/p4/simplifyDefUse.cpp

```diff
@@ -1051,10 +1051,25 @@ class FindUninitialized : public Inspector {
             // This could happen if we are writing to an array element
             // with an unknown index.
             auto type = typeMap->getType(expression, true);
-            warn(ErrorType::WARN_UNINITIALIZED_USE,
-                 type->is<IR::Type_Base>() ? "%1% may be uninitialized"
-                                           : "%1% may not be completely initialized",
-                 expression);
+            if (auto structType = type->to<IR::Type_StructLike>()) {
+                for (auto field : structType->fields) {
+                    auto fieldLoc = read->getField(field->name);
+                    auto fieldPoints = currentDefinitions->getPoints(fieldLoc);
+                    if (fieldPoints->containsBeforeStart()) {
+                        warn(ErrorType::WARN_UNINITIALIZED_USE,
```

Comment:
I just realized this while addressing the comments downstream, but the default error reporter will only print 1 warning per use since the node is the same for each loop iteration. Is that OK? The user having to recompile each time they add a definition for a field isn't the best experience.

Reply:
That can't be true, as some of the changes in expected stderr traces have mulitple fields noted as being uninitialized.

One concern is that a single large struct with many uninitialized fields could trigger a big spew of warnings -- I considered the idea of first finding the uninitialized fields and then emitting a single message based on that (if just a few fields, could list the fields.  Could also detect the corner case when no field is initialized)